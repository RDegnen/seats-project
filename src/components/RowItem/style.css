.seat {
  height: 50px;
  width: 50px;
  display: inline-block;
  margin: 10px 5px;
}

.seat:hover {
  cursor: pointer;
}

.aisle {
  width: 50px;
  display: block;
  margin: 10px 5px;
  font-weight: bold;
  color: grey;
  padding: 15px 0;
}