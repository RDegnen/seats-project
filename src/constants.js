export const classes = {
  FIRST: 'First',
  BUSINESS: 'Business',
  ECONOMY: 'Economy',
}

export const colors = {
  LOLA_BLUE: '#1a60e8',
  LOLA_PINK: '#e83697',
  LIGHT_GREY: 'lightGrey',
}