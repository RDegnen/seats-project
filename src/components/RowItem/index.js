import React from 'react'
import PropTypes from 'prop-types'
import { Col } from 'reactstrap'

import { seatPropType } from '../props'
import { colors } from '../../constants'
import './style.css'

const setColor = (occupied, selected) => ({
  backgroundColor: occupied ? colors.LIGHT_GREY : 
    selected ? colors.LOLA_PINK : colors.LOLA_BLUE
})

const Seat = ({ occupied, onClick, selected }) => (
  <div 
    className="seat" 
    style={setColor(occupied, selected)} 
    onClick={onClick}
  >
  </div>
)

Seat.propTypes = {
  occupied: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
  selected: PropTypes.bool,
}

const Aisle = ({ row }) => (
  <span className="aisle">{row}</span>
)

Aisle.propTypes = {
  row: PropTypes.number.isRequired,
}

const RowItem = ({ item, selectSeat }) => {
  const { aisle, row, occupied, selected } = item
  const handleClick = () => selectSeat(item)
  return (
    <Col>
      {aisle ? (
        <Aisle row={row} />
      ) : (
        <Seat 
          occupied={occupied} 
          onClick={handleClick} 
          selected={selected}
        />
      )}
    </Col>
  )
}

RowItem.propTypes = {
  item: seatPropType.isRequired,
  selectSeat: PropTypes.func.isRequired,
}

export default RowItem