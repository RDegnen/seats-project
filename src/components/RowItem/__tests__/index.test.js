import React from 'react'
import { shallow } from 'enzyme'

import RowItem from '../'
import { colors } from '../../../constants'

const seats = [ 
  { seat: 'A', row: 3, class: 'First', occupied: true },
  { seat: 'B', row: 3, class: 'First', aisle: true },
  { seat: 'C', row: 3, class: 'First', occupied: false },
  { seat: 'D', row: 3, class: 'First', selected: true, occupied: false },
]

const getComponent = props => 
  shallow(<RowItem {...props} selectSeat={jest.fn()} />)

describe('<RowItem />', () => {
  seats.forEach(seat => {
    it('should match snapshot', () => {
      const component = getComponent({ item: seat })
      expect(component).toMatchSnapshot()
    })
  })

  it('should render an Aisle', () => {
    const component = getComponent({ item: seats[1] })
    expect(component.find('Aisle')).toBeTruthy()
  })

  it('should properly color a seat', () => {
    const occupiedSeat = getComponent({ item: seats[0] })
    const availableSeat = getComponent({ item: seats[2] })
    const selectedSeat = getComponent({ item: seats[3] })
    expect(
      occupiedSeat
        .find('Seat')
        .dive()
        .prop('style')
      ).toHaveProperty('backgroundColor', colors.LIGHT_GREY)

    expect(
      availableSeat
        .find('Seat')
        .dive()
        .prop('style')
      ).toHaveProperty('backgroundColor', colors.LOLA_BLUE)

    expect(
      selectedSeat
        .find('Seat')
        .dive()
        .prop('style')
      ).toHaveProperty('backgroundColor', colors.LOLA_PINK)
  })
})