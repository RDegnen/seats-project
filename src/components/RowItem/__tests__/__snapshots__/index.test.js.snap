// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RowItem /> should match snapshot 1`] = `
<Col
  tag="div"
  widths={
    Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ]
  }
>
  <Seat
    occupied={true}
    onClick={[Function]}
  />
</Col>
`;

exports[`<RowItem /> should match snapshot 2`] = `
<Col
  tag="div"
  widths={
    Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ]
  }
>
  <Aisle
    row={3}
  />
</Col>
`;

exports[`<RowItem /> should match snapshot 3`] = `
<Col
  tag="div"
  widths={
    Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ]
  }
>
  <Seat
    occupied={false}
    onClick={[Function]}
  />
</Col>
`;

exports[`<RowItem /> should match snapshot 4`] = `
<Col
  tag="div"
  widths={
    Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ]
  }
>
  <Seat
    occupied={false}
    onClick={[Function]}
    selected={true}
  />
</Col>
`;
